{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "db1c0661",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(rtracklayer)\n",
    "library(RColorBrewer)\n",
    "library(GenomicRanges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1ea24e08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3799629 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>start</th><th scope=col>end</th><th scope=col>width</th><th scope=col>strand</th><th scope=col>source</th><th scope=col>type</th><th scope=col>score</th><th scope=col>phase</th><th scope=col>gene_id</th><th scope=col>transcript_id</th><th scope=col>family_id</th><th scope=col>class_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>  8387807</td><td>  8388657</td><td> 851</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 3777</td><td>NA</td><td>Lx2B2      </td><td>Lx2B2          </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 41942995</td><td> 41943142</td><td> 148</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  595</td><td>NA</td><td>B3         </td><td>B3             </td><td>B2       </td><td>SINE</td></tr>\n",
       "\t<tr><td>1</td><td> 50331619</td><td> 50332377</td><td> 759</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 1796</td><td>NA</td><td>Lx7        </td><td>Lx7            </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 58720078</td><td> 58721182</td><td>1105</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 5180</td><td>NA</td><td>L1MdV_III  </td><td>L1MdV_III      </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td>100663165</td><td>100663479</td><td> 315</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 1316</td><td>NA</td><td>MLTR14     </td><td>MLTR14         </td><td>ERV1     </td><td>LTR </td></tr>\n",
       "\t<tr><td>1</td><td>109051878</td><td>109052234</td><td> 357</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 2314</td><td>NA</td><td>Lx4B       </td><td>Lx4B           </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td>117440147</td><td>117440529</td><td> 383</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 2330</td><td>NA</td><td>Lx2B       </td><td>Lx2B           </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td>142606236</td><td>142606423</td><td> 188</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  328</td><td>NA</td><td>X9_LINE    </td><td>X9_LINE        </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td>176160666</td><td>176160805</td><td> 140</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  560</td><td>NA</td><td>B1F1       </td><td>B1F1           </td><td>Alu      </td><td>SINE</td></tr>\n",
       "\t<tr><td>1</td><td>  3145577</td><td>  3147149</td><td>1573</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 5715</td><td>NA</td><td>Lx9        </td><td>Lx9            </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td>  5242462</td><td>  5242883</td><td> 422</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 1458</td><td>NA</td><td>MuRRS4-int </td><td>MuRRS4-int     </td><td>ERV1     </td><td>LTR </td></tr>\n",
       "\t<tr><td>1</td><td>  6291439</td><td>  6291577</td><td> 139</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  911</td><td>NA</td><td>B1_Mus1    </td><td>B1_Mus1        </td><td>Alu      </td><td>SINE</td></tr>\n",
       "\t<tr><td>1</td><td>  9436574</td><td>  9437274</td><td> 701</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>19049</td><td>NA</td><td>Lx3_Mus    </td><td>Lx3_Mus        </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 10483573</td><td> 10486011</td><td>2439</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>25855</td><td>NA</td><td>L1Lx_IV    </td><td>L1Lx_IV        </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 12582862</td><td> 12584935</td><td>2074</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 8254</td><td>NA</td><td>L1MdF_V    </td><td>L1MdF_V        </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 13631480</td><td> 13631575</td><td>  96</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  344</td><td>NA</td><td>PB1D10     </td><td>PB1D10         </td><td>Alu      </td><td>SINE</td></tr>\n",
       "\t<tr><td>1</td><td> 17825618</td><td> 17825906</td><td> 289</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  194</td><td>NA</td><td>L2         </td><td>L2             </td><td>L2       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 18873499</td><td> 18874429</td><td> 931</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 8074</td><td>NA</td><td>L1MdGf_II  </td><td>L1MdGf_II      </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 24115495</td><td> 24119716</td><td>4222</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>37990</td><td>NA</td><td>L1MdTf_III </td><td>L1MdTf_III     </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 26211780</td><td> 26216546</td><td>4767</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>32457</td><td>NA</td><td>L1MdF_IV   </td><td>L1MdF_IV       </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 27262832</td><td> 27263196</td><td> 365</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  951</td><td>NA</td><td>RMER15     </td><td>RMER15         </td><td>ERVL     </td><td>LTR </td></tr>\n",
       "\t<tr><td>1</td><td> 28311212</td><td> 28311651</td><td> 440</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 3097</td><td>NA</td><td>L1MdFanc_II</td><td>L1MdFanc_II    </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 29359887</td><td> 29360190</td><td> 304</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 1438</td><td>NA</td><td>Lx3B       </td><td>Lx3B           </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 30408619</td><td> 30408747</td><td> 129</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  239</td><td>NA</td><td>MLT1E      </td><td>MLT1E          </td><td>ERVL-MaLR</td><td>LTR </td></tr>\n",
       "\t<tr><td>1</td><td> 35651363</td><td> 35651652</td><td> 290</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  279</td><td>NA</td><td>L1MC       </td><td>L1MC           </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 39845823</td><td> 39845939</td><td> 117</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  349</td><td>NA</td><td>B4A        </td><td>B4A            </td><td>B4       </td><td>SINE</td></tr>\n",
       "\t<tr><td>1</td><td> 44040128</td><td> 44040383</td><td> 256</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  574</td><td>NA</td><td>B4A        </td><td>B4A_dup1       </td><td>B4       </td><td>SINE</td></tr>\n",
       "\t<tr><td>1</td><td> 45088719</td><td> 45089335</td><td> 617</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 5348</td><td>NA</td><td>L1MdF_II   </td><td>L1MdF_II       </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 47182400</td><td> 47188854</td><td>6455</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>38406</td><td>NA</td><td>L1MdTf_III </td><td>L1MdTf_III_dup1</td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>1</td><td> 48231490</td><td> 48237666</td><td>6177</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>38099</td><td>NA</td><td>L1MdF_II   </td><td>L1MdF_II_dup1  </td><td>L1       </td><td>LINE</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>134274</td><td>134451</td><td> 178</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  547</td><td>NA</td><td>MTD        </td><td>MTD_dup54577       </td><td>ERVL-MaLR   </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>135686</td><td>135818</td><td> 133</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  553</td><td>NA</td><td>RMER4A     </td><td>RMER4A_dup3353     </td><td>ERVK        </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>135854</td><td>135975</td><td> 122</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  481</td><td>NA</td><td>RMER1C     </td><td>RMER1C_dup6327     </td><td>L1-dep      </td><td>Retroposon</td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>136254</td><td>136777</td><td> 524</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 2345</td><td>NA</td><td>RLTR45     </td><td>RLTR45_dup1390     </td><td>ERVK        </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>138481</td><td>138588</td><td> 108</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  406</td><td>NA</td><td>MTB        </td><td>MTB_dup8337        </td><td>ERVL-MaLR   </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>138678</td><td>138765</td><td>  88</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  440</td><td>NA</td><td>Lx2        </td><td>Lx2_dup18102       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>138756</td><td>138823</td><td>  68</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  412</td><td>NA</td><td>Lx2        </td><td>Lx2_dup18103       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>138827</td><td>140277</td><td>1451</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>11011</td><td>NA</td><td>ETnERV2-int</td><td>ETnERV2-int_dup2089</td><td>ERVK        </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>140377</td><td>140542</td><td> 166</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  457</td><td>NA</td><td>MTB        </td><td>MTB_dup8338        </td><td>ERVL-MaLR   </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>140548</td><td>140628</td><td>  81</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  234</td><td>NA</td><td>Lx2        </td><td>Lx2_dup18104       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>141081</td><td>141227</td><td> 147</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  706</td><td>NA</td><td>L1MdV_III  </td><td>L1MdV_III_dup11775 </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>141229</td><td>141494</td><td> 266</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 1145</td><td>NA</td><td>L1Lx_III   </td><td>L1Lx_III_dup14441  </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>141554</td><td>141695</td><td> 142</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  819</td><td>NA</td><td>B1_Mus2    </td><td>B1_Mus2_dup70142   </td><td>Alu         </td><td>SINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>141899</td><td>142644</td><td> 746</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 2976</td><td>NA</td><td>CYRA11_Mm  </td><td>CYRA11_Mm_dup4263  </td><td>Y-chromosome</td><td>Satellite </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>142972</td><td>143642</td><td> 671</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 1493</td><td>NA</td><td>Lx7        </td><td>Lx7_dup34457       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>145570</td><td>146105</td><td> 536</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 3905</td><td>NA</td><td>L1MdF_III  </td><td>L1MdF_III_dup13054 </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>146196</td><td>146365</td><td> 170</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  381</td><td>NA</td><td>Lx5b       </td><td>Lx5b_dup8267       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>148278</td><td>148417</td><td> 140</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  658</td><td>NA</td><td>B1_Mus1    </td><td>B1_Mus1_dup96510   </td><td>Alu         </td><td>SINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>148643</td><td>148736</td><td>  94</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  227</td><td>NA</td><td>L1MD       </td><td>L1MD_dup4877       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>148834</td><td>149146</td><td> 313</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 1088</td><td>NA</td><td>MT2B1      </td><td>MT2B1_dup8115      </td><td>ERVL        </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>149535</td><td>149672</td><td> 138</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  579</td><td>NA</td><td>B1_Mus1    </td><td>B1_Mus1_dup96511   </td><td>Alu         </td><td>SINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>149887</td><td>149982</td><td>  96</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  247</td><td>NA</td><td>MurSatRep1 </td><td>MurSatRep1_dup2850 </td><td>MurSatRep1  </td><td>Unknown   </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>150798</td><td>151072</td><td> 275</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 1429</td><td>NA</td><td>L1MdV_III  </td><td>L1MdV_III_dup11776 </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>151266</td><td>151612</td><td> 347</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  708</td><td>NA</td><td>Lx8        </td><td>Lx8_dup57159       </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>151645</td><td>151776</td><td> 132</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>  327</td><td>NA</td><td>MER2       </td><td>MER2_dup10475      </td><td>TcMar-Tigger</td><td>DNA       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>152262</td><td>152508</td><td> 247</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td> 1950</td><td>NA</td><td>L1MdF_II   </td><td>L1MdF_II_dup4437   </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>152557</td><td>155890</td><td>3334</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td>26026</td><td>NA</td><td>L1MdGf_II  </td><td>L1MdGf_II_dup7962  </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>155891</td><td>156898</td><td>1008</td><td>+</td><td>mm39_rmsk</td><td>exon</td><td> 8022</td><td>NA</td><td>L1MdGf_II  </td><td>L1MdGf_II_dup7963  </td><td>L1          </td><td>LINE      </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>157070</td><td>157145</td><td>  76</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  240</td><td>NA</td><td>ORR1E      </td><td>ORR1E_dup25377     </td><td>ERVL-MaLR   </td><td>LTR       </td></tr>\n",
       "\t<tr><td>JH584303.1</td><td>157909</td><td>158099</td><td> 191</td><td>-</td><td>mm39_rmsk</td><td>exon</td><td>  528</td><td>NA</td><td>ORR1A4     </td><td>ORR1A4_dup8610     </td><td>ERVL-MaLR   </td><td>LTR       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3799629 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " seqnames & start & end & width & strand & source & type & score & phase & gene\\_id & transcript\\_id & family\\_id & class\\_id\\\\\n",
       " <fct> & <int> & <int> & <int> & <fct> & <fct> & <fct> & <dbl> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 &   8387807 &   8388657 &  851 & + & mm39\\_rmsk & exon &  3777 & NA & Lx2B2       & Lx2B2           & L1        & LINE\\\\\n",
       "\t 1 &  41942995 &  41943142 &  148 & + & mm39\\_rmsk & exon &   595 & NA & B3          & B3              & B2        & SINE\\\\\n",
       "\t 1 &  50331619 &  50332377 &  759 & - & mm39\\_rmsk & exon &  1796 & NA & Lx7         & Lx7             & L1        & LINE\\\\\n",
       "\t 1 &  58720078 &  58721182 & 1105 & + & mm39\\_rmsk & exon &  5180 & NA & L1MdV\\_III   & L1MdV\\_III       & L1        & LINE\\\\\n",
       "\t 1 & 100663165 & 100663479 &  315 & + & mm39\\_rmsk & exon &  1316 & NA & MLTR14      & MLTR14          & ERV1      & LTR \\\\\n",
       "\t 1 & 109051878 & 109052234 &  357 & - & mm39\\_rmsk & exon &  2314 & NA & Lx4B        & Lx4B            & L1        & LINE\\\\\n",
       "\t 1 & 117440147 & 117440529 &  383 & - & mm39\\_rmsk & exon &  2330 & NA & Lx2B        & Lx2B            & L1        & LINE\\\\\n",
       "\t 1 & 142606236 & 142606423 &  188 & + & mm39\\_rmsk & exon &   328 & NA & X9\\_LINE     & X9\\_LINE         & L1        & LINE\\\\\n",
       "\t 1 & 176160666 & 176160805 &  140 & + & mm39\\_rmsk & exon &   560 & NA & B1F1        & B1F1            & Alu       & SINE\\\\\n",
       "\t 1 &   3145577 &   3147149 & 1573 & + & mm39\\_rmsk & exon &  5715 & NA & Lx9         & Lx9             & L1        & LINE\\\\\n",
       "\t 1 &   5242462 &   5242883 &  422 & + & mm39\\_rmsk & exon &  1458 & NA & MuRRS4-int  & MuRRS4-int      & ERV1      & LTR \\\\\n",
       "\t 1 &   6291439 &   6291577 &  139 & - & mm39\\_rmsk & exon &   911 & NA & B1\\_Mus1     & B1\\_Mus1         & Alu       & SINE\\\\\n",
       "\t 1 &   9436574 &   9437274 &  701 & - & mm39\\_rmsk & exon & 19049 & NA & Lx3\\_Mus     & Lx3\\_Mus         & L1        & LINE\\\\\n",
       "\t 1 &  10483573 &  10486011 & 2439 & - & mm39\\_rmsk & exon & 25855 & NA & L1Lx\\_IV     & L1Lx\\_IV         & L1        & LINE\\\\\n",
       "\t 1 &  12582862 &  12584935 & 2074 & + & mm39\\_rmsk & exon &  8254 & NA & L1MdF\\_V     & L1MdF\\_V         & L1        & LINE\\\\\n",
       "\t 1 &  13631480 &  13631575 &   96 & + & mm39\\_rmsk & exon &   344 & NA & PB1D10      & PB1D10          & Alu       & SINE\\\\\n",
       "\t 1 &  17825618 &  17825906 &  289 & - & mm39\\_rmsk & exon &   194 & NA & L2          & L2              & L2        & LINE\\\\\n",
       "\t 1 &  18873499 &  18874429 &  931 & + & mm39\\_rmsk & exon &  8074 & NA & L1MdGf\\_II   & L1MdGf\\_II       & L1        & LINE\\\\\n",
       "\t 1 &  24115495 &  24119716 & 4222 & + & mm39\\_rmsk & exon & 37990 & NA & L1MdTf\\_III  & L1MdTf\\_III      & L1        & LINE\\\\\n",
       "\t 1 &  26211780 &  26216546 & 4767 & + & mm39\\_rmsk & exon & 32457 & NA & L1MdF\\_IV    & L1MdF\\_IV        & L1        & LINE\\\\\n",
       "\t 1 &  27262832 &  27263196 &  365 & + & mm39\\_rmsk & exon &   951 & NA & RMER15      & RMER15          & ERVL      & LTR \\\\\n",
       "\t 1 &  28311212 &  28311651 &  440 & + & mm39\\_rmsk & exon &  3097 & NA & L1MdFanc\\_II & L1MdFanc\\_II     & L1        & LINE\\\\\n",
       "\t 1 &  29359887 &  29360190 &  304 & - & mm39\\_rmsk & exon &  1438 & NA & Lx3B        & Lx3B            & L1        & LINE\\\\\n",
       "\t 1 &  30408619 &  30408747 &  129 & + & mm39\\_rmsk & exon &   239 & NA & MLT1E       & MLT1E           & ERVL-MaLR & LTR \\\\\n",
       "\t 1 &  35651363 &  35651652 &  290 & + & mm39\\_rmsk & exon &   279 & NA & L1MC        & L1MC            & L1        & LINE\\\\\n",
       "\t 1 &  39845823 &  39845939 &  117 & + & mm39\\_rmsk & exon &   349 & NA & B4A         & B4A             & B4        & SINE\\\\\n",
       "\t 1 &  44040128 &  44040383 &  256 & + & mm39\\_rmsk & exon &   574 & NA & B4A         & B4A\\_dup1        & B4        & SINE\\\\\n",
       "\t 1 &  45088719 &  45089335 &  617 & - & mm39\\_rmsk & exon &  5348 & NA & L1MdF\\_II    & L1MdF\\_II        & L1        & LINE\\\\\n",
       "\t 1 &  47182400 &  47188854 & 6455 & - & mm39\\_rmsk & exon & 38406 & NA & L1MdTf\\_III  & L1MdTf\\_III\\_dup1 & L1        & LINE\\\\\n",
       "\t 1 &  48231490 &  48237666 & 6177 & - & mm39\\_rmsk & exon & 38099 & NA & L1MdF\\_II    & L1MdF\\_II\\_dup1   & L1        & LINE\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t JH584303.1 & 134274 & 134451 &  178 & + & mm39\\_rmsk & exon &   547 & NA & MTD         & MTD\\_dup54577        & ERVL-MaLR    & LTR       \\\\\n",
       "\t JH584303.1 & 135686 & 135818 &  133 & + & mm39\\_rmsk & exon &   553 & NA & RMER4A      & RMER4A\\_dup3353      & ERVK         & LTR       \\\\\n",
       "\t JH584303.1 & 135854 & 135975 &  122 & - & mm39\\_rmsk & exon &   481 & NA & RMER1C      & RMER1C\\_dup6327      & L1-dep       & Retroposon\\\\\n",
       "\t JH584303.1 & 136254 & 136777 &  524 & + & mm39\\_rmsk & exon &  2345 & NA & RLTR45      & RLTR45\\_dup1390      & ERVK         & LTR       \\\\\n",
       "\t JH584303.1 & 138481 & 138588 &  108 & + & mm39\\_rmsk & exon &   406 & NA & MTB         & MTB\\_dup8337         & ERVL-MaLR    & LTR       \\\\\n",
       "\t JH584303.1 & 138678 & 138765 &   88 & + & mm39\\_rmsk & exon &   440 & NA & Lx2         & Lx2\\_dup18102        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 138756 & 138823 &   68 & + & mm39\\_rmsk & exon &   412 & NA & Lx2         & Lx2\\_dup18103        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 138827 & 140277 & 1451 & - & mm39\\_rmsk & exon & 11011 & NA & ETnERV2-int & ETnERV2-int\\_dup2089 & ERVK         & LTR       \\\\\n",
       "\t JH584303.1 & 140377 & 140542 &  166 & + & mm39\\_rmsk & exon &   457 & NA & MTB         & MTB\\_dup8338         & ERVL-MaLR    & LTR       \\\\\n",
       "\t JH584303.1 & 140548 & 140628 &   81 & + & mm39\\_rmsk & exon &   234 & NA & Lx2         & Lx2\\_dup18104        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 141081 & 141227 &  147 & + & mm39\\_rmsk & exon &   706 & NA & L1MdV\\_III   & L1MdV\\_III\\_dup11775  & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 141229 & 141494 &  266 & + & mm39\\_rmsk & exon &  1145 & NA & L1Lx\\_III    & L1Lx\\_III\\_dup14441   & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 141554 & 141695 &  142 & + & mm39\\_rmsk & exon &   819 & NA & B1\\_Mus2     & B1\\_Mus2\\_dup70142    & Alu          & SINE      \\\\\n",
       "\t JH584303.1 & 141899 & 142644 &  746 & + & mm39\\_rmsk & exon &  2976 & NA & CYRA11\\_Mm   & CYRA11\\_Mm\\_dup4263   & Y-chromosome & Satellite \\\\\n",
       "\t JH584303.1 & 142972 & 143642 &  671 & - & mm39\\_rmsk & exon &  1493 & NA & Lx7         & Lx7\\_dup34457        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 145570 & 146105 &  536 & + & mm39\\_rmsk & exon &  3905 & NA & L1MdF\\_III   & L1MdF\\_III\\_dup13054  & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 146196 & 146365 &  170 & + & mm39\\_rmsk & exon &   381 & NA & Lx5b        & Lx5b\\_dup8267        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 148278 & 148417 &  140 & - & mm39\\_rmsk & exon &   658 & NA & B1\\_Mus1     & B1\\_Mus1\\_dup96510    & Alu          & SINE      \\\\\n",
       "\t JH584303.1 & 148643 & 148736 &   94 & + & mm39\\_rmsk & exon &   227 & NA & L1MD        & L1MD\\_dup4877        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 148834 & 149146 &  313 & + & mm39\\_rmsk & exon &  1088 & NA & MT2B1       & MT2B1\\_dup8115       & ERVL         & LTR       \\\\\n",
       "\t JH584303.1 & 149535 & 149672 &  138 & + & mm39\\_rmsk & exon &   579 & NA & B1\\_Mus1     & B1\\_Mus1\\_dup96511    & Alu          & SINE      \\\\\n",
       "\t JH584303.1 & 149887 & 149982 &   96 & - & mm39\\_rmsk & exon &   247 & NA & MurSatRep1  & MurSatRep1\\_dup2850  & MurSatRep1   & Unknown   \\\\\n",
       "\t JH584303.1 & 150798 & 151072 &  275 & + & mm39\\_rmsk & exon &  1429 & NA & L1MdV\\_III   & L1MdV\\_III\\_dup11776  & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 151266 & 151612 &  347 & + & mm39\\_rmsk & exon &   708 & NA & Lx8         & Lx8\\_dup57159        & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 151645 & 151776 &  132 & + & mm39\\_rmsk & exon &   327 & NA & MER2        & MER2\\_dup10475       & TcMar-Tigger & DNA       \\\\\n",
       "\t JH584303.1 & 152262 & 152508 &  247 & - & mm39\\_rmsk & exon &  1950 & NA & L1MdF\\_II    & L1MdF\\_II\\_dup4437    & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 152557 & 155890 & 3334 & + & mm39\\_rmsk & exon & 26026 & NA & L1MdGf\\_II   & L1MdGf\\_II\\_dup7962   & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 155891 & 156898 & 1008 & + & mm39\\_rmsk & exon &  8022 & NA & L1MdGf\\_II   & L1MdGf\\_II\\_dup7963   & L1           & LINE      \\\\\n",
       "\t JH584303.1 & 157070 & 157145 &   76 & - & mm39\\_rmsk & exon &   240 & NA & ORR1E       & ORR1E\\_dup25377      & ERVL-MaLR    & LTR       \\\\\n",
       "\t JH584303.1 & 157909 & 158099 &  191 & - & mm39\\_rmsk & exon &   528 & NA & ORR1A4      & ORR1A4\\_dup8610      & ERVL-MaLR    & LTR       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3799629 × 13\n",
       "\n",
       "| seqnames &lt;fct&gt; | start &lt;int&gt; | end &lt;int&gt; | width &lt;int&gt; | strand &lt;fct&gt; | source &lt;fct&gt; | type &lt;fct&gt; | score &lt;dbl&gt; | phase &lt;int&gt; | gene_id &lt;chr&gt; | transcript_id &lt;chr&gt; | family_id &lt;chr&gt; | class_id &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 |   8387807 |   8388657 |  851 | + | mm39_rmsk | exon |  3777 | NA | Lx2B2       | Lx2B2           | L1        | LINE |\n",
       "| 1 |  41942995 |  41943142 |  148 | + | mm39_rmsk | exon |   595 | NA | B3          | B3              | B2        | SINE |\n",
       "| 1 |  50331619 |  50332377 |  759 | - | mm39_rmsk | exon |  1796 | NA | Lx7         | Lx7             | L1        | LINE |\n",
       "| 1 |  58720078 |  58721182 | 1105 | + | mm39_rmsk | exon |  5180 | NA | L1MdV_III   | L1MdV_III       | L1        | LINE |\n",
       "| 1 | 100663165 | 100663479 |  315 | + | mm39_rmsk | exon |  1316 | NA | MLTR14      | MLTR14          | ERV1      | LTR  |\n",
       "| 1 | 109051878 | 109052234 |  357 | - | mm39_rmsk | exon |  2314 | NA | Lx4B        | Lx4B            | L1        | LINE |\n",
       "| 1 | 117440147 | 117440529 |  383 | - | mm39_rmsk | exon |  2330 | NA | Lx2B        | Lx2B            | L1        | LINE |\n",
       "| 1 | 142606236 | 142606423 |  188 | + | mm39_rmsk | exon |   328 | NA | X9_LINE     | X9_LINE         | L1        | LINE |\n",
       "| 1 | 176160666 | 176160805 |  140 | + | mm39_rmsk | exon |   560 | NA | B1F1        | B1F1            | Alu       | SINE |\n",
       "| 1 |   3145577 |   3147149 | 1573 | + | mm39_rmsk | exon |  5715 | NA | Lx9         | Lx9             | L1        | LINE |\n",
       "| 1 |   5242462 |   5242883 |  422 | + | mm39_rmsk | exon |  1458 | NA | MuRRS4-int  | MuRRS4-int      | ERV1      | LTR  |\n",
       "| 1 |   6291439 |   6291577 |  139 | - | mm39_rmsk | exon |   911 | NA | B1_Mus1     | B1_Mus1         | Alu       | SINE |\n",
       "| 1 |   9436574 |   9437274 |  701 | - | mm39_rmsk | exon | 19049 | NA | Lx3_Mus     | Lx3_Mus         | L1        | LINE |\n",
       "| 1 |  10483573 |  10486011 | 2439 | - | mm39_rmsk | exon | 25855 | NA | L1Lx_IV     | L1Lx_IV         | L1        | LINE |\n",
       "| 1 |  12582862 |  12584935 | 2074 | + | mm39_rmsk | exon |  8254 | NA | L1MdF_V     | L1MdF_V         | L1        | LINE |\n",
       "| 1 |  13631480 |  13631575 |   96 | + | mm39_rmsk | exon |   344 | NA | PB1D10      | PB1D10          | Alu       | SINE |\n",
       "| 1 |  17825618 |  17825906 |  289 | - | mm39_rmsk | exon |   194 | NA | L2          | L2              | L2        | LINE |\n",
       "| 1 |  18873499 |  18874429 |  931 | + | mm39_rmsk | exon |  8074 | NA | L1MdGf_II   | L1MdGf_II       | L1        | LINE |\n",
       "| 1 |  24115495 |  24119716 | 4222 | + | mm39_rmsk | exon | 37990 | NA | L1MdTf_III  | L1MdTf_III      | L1        | LINE |\n",
       "| 1 |  26211780 |  26216546 | 4767 | + | mm39_rmsk | exon | 32457 | NA | L1MdF_IV    | L1MdF_IV        | L1        | LINE |\n",
       "| 1 |  27262832 |  27263196 |  365 | + | mm39_rmsk | exon |   951 | NA | RMER15      | RMER15          | ERVL      | LTR  |\n",
       "| 1 |  28311212 |  28311651 |  440 | + | mm39_rmsk | exon |  3097 | NA | L1MdFanc_II | L1MdFanc_II     | L1        | LINE |\n",
       "| 1 |  29359887 |  29360190 |  304 | - | mm39_rmsk | exon |  1438 | NA | Lx3B        | Lx3B            | L1        | LINE |\n",
       "| 1 |  30408619 |  30408747 |  129 | + | mm39_rmsk | exon |   239 | NA | MLT1E       | MLT1E           | ERVL-MaLR | LTR  |\n",
       "| 1 |  35651363 |  35651652 |  290 | + | mm39_rmsk | exon |   279 | NA | L1MC        | L1MC            | L1        | LINE |\n",
       "| 1 |  39845823 |  39845939 |  117 | + | mm39_rmsk | exon |   349 | NA | B4A         | B4A             | B4        | SINE |\n",
       "| 1 |  44040128 |  44040383 |  256 | + | mm39_rmsk | exon |   574 | NA | B4A         | B4A_dup1        | B4        | SINE |\n",
       "| 1 |  45088719 |  45089335 |  617 | - | mm39_rmsk | exon |  5348 | NA | L1MdF_II    | L1MdF_II        | L1        | LINE |\n",
       "| 1 |  47182400 |  47188854 | 6455 | - | mm39_rmsk | exon | 38406 | NA | L1MdTf_III  | L1MdTf_III_dup1 | L1        | LINE |\n",
       "| 1 |  48231490 |  48237666 | 6177 | - | mm39_rmsk | exon | 38099 | NA | L1MdF_II    | L1MdF_II_dup1   | L1        | LINE |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| JH584303.1 | 134274 | 134451 |  178 | + | mm39_rmsk | exon |   547 | NA | MTD         | MTD_dup54577        | ERVL-MaLR    | LTR        |\n",
       "| JH584303.1 | 135686 | 135818 |  133 | + | mm39_rmsk | exon |   553 | NA | RMER4A      | RMER4A_dup3353      | ERVK         | LTR        |\n",
       "| JH584303.1 | 135854 | 135975 |  122 | - | mm39_rmsk | exon |   481 | NA | RMER1C      | RMER1C_dup6327      | L1-dep       | Retroposon |\n",
       "| JH584303.1 | 136254 | 136777 |  524 | + | mm39_rmsk | exon |  2345 | NA | RLTR45      | RLTR45_dup1390      | ERVK         | LTR        |\n",
       "| JH584303.1 | 138481 | 138588 |  108 | + | mm39_rmsk | exon |   406 | NA | MTB         | MTB_dup8337         | ERVL-MaLR    | LTR        |\n",
       "| JH584303.1 | 138678 | 138765 |   88 | + | mm39_rmsk | exon |   440 | NA | Lx2         | Lx2_dup18102        | L1           | LINE       |\n",
       "| JH584303.1 | 138756 | 138823 |   68 | + | mm39_rmsk | exon |   412 | NA | Lx2         | Lx2_dup18103        | L1           | LINE       |\n",
       "| JH584303.1 | 138827 | 140277 | 1451 | - | mm39_rmsk | exon | 11011 | NA | ETnERV2-int | ETnERV2-int_dup2089 | ERVK         | LTR        |\n",
       "| JH584303.1 | 140377 | 140542 |  166 | + | mm39_rmsk | exon |   457 | NA | MTB         | MTB_dup8338         | ERVL-MaLR    | LTR        |\n",
       "| JH584303.1 | 140548 | 140628 |   81 | + | mm39_rmsk | exon |   234 | NA | Lx2         | Lx2_dup18104        | L1           | LINE       |\n",
       "| JH584303.1 | 141081 | 141227 |  147 | + | mm39_rmsk | exon |   706 | NA | L1MdV_III   | L1MdV_III_dup11775  | L1           | LINE       |\n",
       "| JH584303.1 | 141229 | 141494 |  266 | + | mm39_rmsk | exon |  1145 | NA | L1Lx_III    | L1Lx_III_dup14441   | L1           | LINE       |\n",
       "| JH584303.1 | 141554 | 141695 |  142 | + | mm39_rmsk | exon |   819 | NA | B1_Mus2     | B1_Mus2_dup70142    | Alu          | SINE       |\n",
       "| JH584303.1 | 141899 | 142644 |  746 | + | mm39_rmsk | exon |  2976 | NA | CYRA11_Mm   | CYRA11_Mm_dup4263   | Y-chromosome | Satellite  |\n",
       "| JH584303.1 | 142972 | 143642 |  671 | - | mm39_rmsk | exon |  1493 | NA | Lx7         | Lx7_dup34457        | L1           | LINE       |\n",
       "| JH584303.1 | 145570 | 146105 |  536 | + | mm39_rmsk | exon |  3905 | NA | L1MdF_III   | L1MdF_III_dup13054  | L1           | LINE       |\n",
       "| JH584303.1 | 146196 | 146365 |  170 | + | mm39_rmsk | exon |   381 | NA | Lx5b        | Lx5b_dup8267        | L1           | LINE       |\n",
       "| JH584303.1 | 148278 | 148417 |  140 | - | mm39_rmsk | exon |   658 | NA | B1_Mus1     | B1_Mus1_dup96510    | Alu          | SINE       |\n",
       "| JH584303.1 | 148643 | 148736 |   94 | + | mm39_rmsk | exon |   227 | NA | L1MD        | L1MD_dup4877        | L1           | LINE       |\n",
       "| JH584303.1 | 148834 | 149146 |  313 | + | mm39_rmsk | exon |  1088 | NA | MT2B1       | MT2B1_dup8115       | ERVL         | LTR        |\n",
       "| JH584303.1 | 149535 | 149672 |  138 | + | mm39_rmsk | exon |   579 | NA | B1_Mus1     | B1_Mus1_dup96511    | Alu          | SINE       |\n",
       "| JH584303.1 | 149887 | 149982 |   96 | - | mm39_rmsk | exon |   247 | NA | MurSatRep1  | MurSatRep1_dup2850  | MurSatRep1   | Unknown    |\n",
       "| JH584303.1 | 150798 | 151072 |  275 | + | mm39_rmsk | exon |  1429 | NA | L1MdV_III   | L1MdV_III_dup11776  | L1           | LINE       |\n",
       "| JH584303.1 | 151266 | 151612 |  347 | + | mm39_rmsk | exon |   708 | NA | Lx8         | Lx8_dup57159        | L1           | LINE       |\n",
       "| JH584303.1 | 151645 | 151776 |  132 | + | mm39_rmsk | exon |   327 | NA | MER2        | MER2_dup10475       | TcMar-Tigger | DNA        |\n",
       "| JH584303.1 | 152262 | 152508 |  247 | - | mm39_rmsk | exon |  1950 | NA | L1MdF_II    | L1MdF_II_dup4437    | L1           | LINE       |\n",
       "| JH584303.1 | 152557 | 155890 | 3334 | + | mm39_rmsk | exon | 26026 | NA | L1MdGf_II   | L1MdGf_II_dup7962   | L1           | LINE       |\n",
       "| JH584303.1 | 155891 | 156898 | 1008 | + | mm39_rmsk | exon |  8022 | NA | L1MdGf_II   | L1MdGf_II_dup7963   | L1           | LINE       |\n",
       "| JH584303.1 | 157070 | 157145 |   76 | - | mm39_rmsk | exon |   240 | NA | ORR1E       | ORR1E_dup25377      | ERVL-MaLR    | LTR        |\n",
       "| JH584303.1 | 157909 | 158099 |  191 | - | mm39_rmsk | exon |   528 | NA | ORR1A4      | ORR1A4_dup8610      | ERVL-MaLR    | LTR        |\n",
       "\n"
      ],
      "text/plain": [
       "        seqnames   start     end       width strand source    type score phase\n",
       "1       1            8387807   8388657  851  +      mm39_rmsk exon  3777 NA   \n",
       "2       1           41942995  41943142  148  +      mm39_rmsk exon   595 NA   \n",
       "3       1           50331619  50332377  759  -      mm39_rmsk exon  1796 NA   \n",
       "4       1           58720078  58721182 1105  +      mm39_rmsk exon  5180 NA   \n",
       "5       1          100663165 100663479  315  +      mm39_rmsk exon  1316 NA   \n",
       "6       1          109051878 109052234  357  -      mm39_rmsk exon  2314 NA   \n",
       "7       1          117440147 117440529  383  -      mm39_rmsk exon  2330 NA   \n",
       "8       1          142606236 142606423  188  +      mm39_rmsk exon   328 NA   \n",
       "9       1          176160666 176160805  140  +      mm39_rmsk exon   560 NA   \n",
       "10      1            3145577   3147149 1573  +      mm39_rmsk exon  5715 NA   \n",
       "11      1            5242462   5242883  422  +      mm39_rmsk exon  1458 NA   \n",
       "12      1            6291439   6291577  139  -      mm39_rmsk exon   911 NA   \n",
       "13      1            9436574   9437274  701  -      mm39_rmsk exon 19049 NA   \n",
       "14      1           10483573  10486011 2439  -      mm39_rmsk exon 25855 NA   \n",
       "15      1           12582862  12584935 2074  +      mm39_rmsk exon  8254 NA   \n",
       "16      1           13631480  13631575   96  +      mm39_rmsk exon   344 NA   \n",
       "17      1           17825618  17825906  289  -      mm39_rmsk exon   194 NA   \n",
       "18      1           18873499  18874429  931  +      mm39_rmsk exon  8074 NA   \n",
       "19      1           24115495  24119716 4222  +      mm39_rmsk exon 37990 NA   \n",
       "20      1           26211780  26216546 4767  +      mm39_rmsk exon 32457 NA   \n",
       "21      1           27262832  27263196  365  +      mm39_rmsk exon   951 NA   \n",
       "22      1           28311212  28311651  440  +      mm39_rmsk exon  3097 NA   \n",
       "23      1           29359887  29360190  304  -      mm39_rmsk exon  1438 NA   \n",
       "24      1           30408619  30408747  129  +      mm39_rmsk exon   239 NA   \n",
       "25      1           35651363  35651652  290  +      mm39_rmsk exon   279 NA   \n",
       "26      1           39845823  39845939  117  +      mm39_rmsk exon   349 NA   \n",
       "27      1           44040128  44040383  256  +      mm39_rmsk exon   574 NA   \n",
       "28      1           45088719  45089335  617  -      mm39_rmsk exon  5348 NA   \n",
       "29      1           47182400  47188854 6455  -      mm39_rmsk exon 38406 NA   \n",
       "30      1           48231490  48237666 6177  -      mm39_rmsk exon 38099 NA   \n",
       "⋮       ⋮          ⋮         ⋮         ⋮     ⋮      ⋮         ⋮    ⋮     ⋮    \n",
       "3799600 JH584303.1 134274    134451     178  +      mm39_rmsk exon   547 NA   \n",
       "3799601 JH584303.1 135686    135818     133  +      mm39_rmsk exon   553 NA   \n",
       "3799602 JH584303.1 135854    135975     122  -      mm39_rmsk exon   481 NA   \n",
       "3799603 JH584303.1 136254    136777     524  +      mm39_rmsk exon  2345 NA   \n",
       "3799604 JH584303.1 138481    138588     108  +      mm39_rmsk exon   406 NA   \n",
       "3799605 JH584303.1 138678    138765      88  +      mm39_rmsk exon   440 NA   \n",
       "3799606 JH584303.1 138756    138823      68  +      mm39_rmsk exon   412 NA   \n",
       "3799607 JH584303.1 138827    140277    1451  -      mm39_rmsk exon 11011 NA   \n",
       "3799608 JH584303.1 140377    140542     166  +      mm39_rmsk exon   457 NA   \n",
       "3799609 JH584303.1 140548    140628      81  +      mm39_rmsk exon   234 NA   \n",
       "3799610 JH584303.1 141081    141227     147  +      mm39_rmsk exon   706 NA   \n",
       "3799611 JH584303.1 141229    141494     266  +      mm39_rmsk exon  1145 NA   \n",
       "3799612 JH584303.1 141554    141695     142  +      mm39_rmsk exon   819 NA   \n",
       "3799613 JH584303.1 141899    142644     746  +      mm39_rmsk exon  2976 NA   \n",
       "3799614 JH584303.1 142972    143642     671  -      mm39_rmsk exon  1493 NA   \n",
       "3799615 JH584303.1 145570    146105     536  +      mm39_rmsk exon  3905 NA   \n",
       "3799616 JH584303.1 146196    146365     170  +      mm39_rmsk exon   381 NA   \n",
       "3799617 JH584303.1 148278    148417     140  -      mm39_rmsk exon   658 NA   \n",
       "3799618 JH584303.1 148643    148736      94  +      mm39_rmsk exon   227 NA   \n",
       "3799619 JH584303.1 148834    149146     313  +      mm39_rmsk exon  1088 NA   \n",
       "3799620 JH584303.1 149535    149672     138  +      mm39_rmsk exon   579 NA   \n",
       "3799621 JH584303.1 149887    149982      96  -      mm39_rmsk exon   247 NA   \n",
       "3799622 JH584303.1 150798    151072     275  +      mm39_rmsk exon  1429 NA   \n",
       "3799623 JH584303.1 151266    151612     347  +      mm39_rmsk exon   708 NA   \n",
       "3799624 JH584303.1 151645    151776     132  +      mm39_rmsk exon   327 NA   \n",
       "3799625 JH584303.1 152262    152508     247  -      mm39_rmsk exon  1950 NA   \n",
       "3799626 JH584303.1 152557    155890    3334  +      mm39_rmsk exon 26026 NA   \n",
       "3799627 JH584303.1 155891    156898    1008  +      mm39_rmsk exon  8022 NA   \n",
       "3799628 JH584303.1 157070    157145      76  -      mm39_rmsk exon   240 NA   \n",
       "3799629 JH584303.1 157909    158099     191  -      mm39_rmsk exon   528 NA   \n",
       "        gene_id     transcript_id       family_id    class_id  \n",
       "1       Lx2B2       Lx2B2               L1           LINE      \n",
       "2       B3          B3                  B2           SINE      \n",
       "3       Lx7         Lx7                 L1           LINE      \n",
       "4       L1MdV_III   L1MdV_III           L1           LINE      \n",
       "5       MLTR14      MLTR14              ERV1         LTR       \n",
       "6       Lx4B        Lx4B                L1           LINE      \n",
       "7       Lx2B        Lx2B                L1           LINE      \n",
       "8       X9_LINE     X9_LINE             L1           LINE      \n",
       "9       B1F1        B1F1                Alu          SINE      \n",
       "10      Lx9         Lx9                 L1           LINE      \n",
       "11      MuRRS4-int  MuRRS4-int          ERV1         LTR       \n",
       "12      B1_Mus1     B1_Mus1             Alu          SINE      \n",
       "13      Lx3_Mus     Lx3_Mus             L1           LINE      \n",
       "14      L1Lx_IV     L1Lx_IV             L1           LINE      \n",
       "15      L1MdF_V     L1MdF_V             L1           LINE      \n",
       "16      PB1D10      PB1D10              Alu          SINE      \n",
       "17      L2          L2                  L2           LINE      \n",
       "18      L1MdGf_II   L1MdGf_II           L1           LINE      \n",
       "19      L1MdTf_III  L1MdTf_III          L1           LINE      \n",
       "20      L1MdF_IV    L1MdF_IV            L1           LINE      \n",
       "21      RMER15      RMER15              ERVL         LTR       \n",
       "22      L1MdFanc_II L1MdFanc_II         L1           LINE      \n",
       "23      Lx3B        Lx3B                L1           LINE      \n",
       "24      MLT1E       MLT1E               ERVL-MaLR    LTR       \n",
       "25      L1MC        L1MC                L1           LINE      \n",
       "26      B4A         B4A                 B4           SINE      \n",
       "27      B4A         B4A_dup1            B4           SINE      \n",
       "28      L1MdF_II    L1MdF_II            L1           LINE      \n",
       "29      L1MdTf_III  L1MdTf_III_dup1     L1           LINE      \n",
       "30      L1MdF_II    L1MdF_II_dup1       L1           LINE      \n",
       "⋮       ⋮           ⋮                   ⋮            ⋮         \n",
       "3799600 MTD         MTD_dup54577        ERVL-MaLR    LTR       \n",
       "3799601 RMER4A      RMER4A_dup3353      ERVK         LTR       \n",
       "3799602 RMER1C      RMER1C_dup6327      L1-dep       Retroposon\n",
       "3799603 RLTR45      RLTR45_dup1390      ERVK         LTR       \n",
       "3799604 MTB         MTB_dup8337         ERVL-MaLR    LTR       \n",
       "3799605 Lx2         Lx2_dup18102        L1           LINE      \n",
       "3799606 Lx2         Lx2_dup18103        L1           LINE      \n",
       "3799607 ETnERV2-int ETnERV2-int_dup2089 ERVK         LTR       \n",
       "3799608 MTB         MTB_dup8338         ERVL-MaLR    LTR       \n",
       "3799609 Lx2         Lx2_dup18104        L1           LINE      \n",
       "3799610 L1MdV_III   L1MdV_III_dup11775  L1           LINE      \n",
       "3799611 L1Lx_III    L1Lx_III_dup14441   L1           LINE      \n",
       "3799612 B1_Mus2     B1_Mus2_dup70142    Alu          SINE      \n",
       "3799613 CYRA11_Mm   CYRA11_Mm_dup4263   Y-chromosome Satellite \n",
       "3799614 Lx7         Lx7_dup34457        L1           LINE      \n",
       "3799615 L1MdF_III   L1MdF_III_dup13054  L1           LINE      \n",
       "3799616 Lx5b        Lx5b_dup8267        L1           LINE      \n",
       "3799617 B1_Mus1     B1_Mus1_dup96510    Alu          SINE      \n",
       "3799618 L1MD        L1MD_dup4877        L1           LINE      \n",
       "3799619 MT2B1       MT2B1_dup8115       ERVL         LTR       \n",
       "3799620 B1_Mus1     B1_Mus1_dup96511    Alu          SINE      \n",
       "3799621 MurSatRep1  MurSatRep1_dup2850  MurSatRep1   Unknown   \n",
       "3799622 L1MdV_III   L1MdV_III_dup11776  L1           LINE      \n",
       "3799623 Lx8         Lx8_dup57159        L1           LINE      \n",
       "3799624 MER2        MER2_dup10475       TcMar-Tigger DNA       \n",
       "3799625 L1MdF_II    L1MdF_II_dup4437    L1           LINE      \n",
       "3799626 L1MdGf_II   L1MdGf_II_dup7962   L1           LINE      \n",
       "3799627 L1MdGf_II   L1MdGf_II_dup7963   L1           LINE      \n",
       "3799628 ORR1E       ORR1E_dup25377      ERVL-MaLR    LTR       \n",
       "3799629 ORR1A4      ORR1A4_dup8610      ERVL-MaLR    LTR       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "TE <- import('/u/home/j/jadiruss/project-clarka/Genomes/mm39_STAR/TE/GRCm39_Ensembl_rmsk_TE.gtf')\n",
    "TE.frame <- as.data.frame(TE)\n",
    "TE.frame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "75340d8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load in MACS3 peakcalls from H3K27ac ChIP-seq datasets\n",
    "cols <- c(\"seqname\", \"Start\", \"End\", \"PeakName\", \"-10*log10pvalue\",\"Unassigned\",\"Summit FC\", \"-log10pvalue\", \"-log10qvalue\", \"SummitPos\")\n",
    "d6_peaks <- read.delim(file = 'd6_M_peaks.narrowPeak', sep = '\\t', header = F)\n",
    "E135_M <- read.delim(file = 'E135_M_peaks.narrowPeak', sep = '\\t', header = F)\n",
    "E135_F <- read.delim(file = 'E135_F_peaks.narrowPeak', sep = '\\t', header = F)\n",
    "colnames(d6_peaks) <- cols\n",
    "colnames(E135_M) <- cols\n",
    "colnames(E135_F) <- cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3bd9db4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load in TEs overlapping with peaks from Fig. 1 ATAC-seq data.\n",
    "cols <- c(\"seqname\", \"Start\", \"End\", \"DupID\", \"GeneID\", \"FamilyID\", \"ClassID\")\n",
    "Male_OC <- read.delim(file = '/u/home/j/jadiruss/project-clarka/TRIM28/ATAC-seq/Allucial_WTonly/Male_OC_WTonly.bed', sep = '\\t', header = F)\n",
    "Male_CO <- read.delim(file = '/u/home/j/jadiruss/project-clarka/TRIM28/ATAC-seq/Allucial_WTonly/Male_CO_WTonly.bed', sep = '\\t', header = F)\n",
    "Female_OC <- read.delim(file = '/u/home/j/jadiruss/project-clarka/TRIM28/ATAC-seq/Allucial_WTonly/Female_OC_WTonly.bed', sep = '\\t', header = F)\n",
    "Female_CO <- read.delim(file = '/u/home/j/jadiruss/project-clarka/TRIM28/ATAC-seq/Allucial_WTonly/Female_CO_WTonly.bed', sep = '\\t', header = F)\n",
    "colnames(Male_OC) <- cols\n",
    "colnames(Male_CO) <- cols\n",
    "colnames(Female_OC) <- cols\n",
    "colnames(Female_CO) <- cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "eaa06955",
   "metadata": {},
   "outputs": [],
   "source": [
    "d6_peak_granges <- GRanges(seqnames=d6_peaks$seqname,ranges=IRanges(start = d6_peaks$Start,end = d6_peaks$End, names = d6_peaks$PeakName),strand=NULL,d6_peaks$`-10*log10pvalue`)\n",
    "E135_M_granges <- GRanges(seqnames=E135_M$seqname,ranges=IRanges(start = E135_M$Start,end = E135_M$End, names = E135_M$PeakName),strand=NULL,E135_M$`-10*log10pvalue`)\n",
    "E135_F_granges <- GRanges(seqnames=E135_F$seqname,ranges=IRanges(start = E135_F$Start,end = E135_F$End, names = E135_F$PeakName),strand=NULL,E135_F$`-10*log10pvalue`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0f9f6a92",
   "metadata": {},
   "outputs": [],
   "source": [
    "Male_OC_granges <- GRanges(seqnames=Male_OC$seqname,ranges=IRanges(start = Male_OC$Start,end = Male_OC$End, names = Male_OC$GeneID),strand=NULL,Male_OC$DupID,Male_OC$FamilyID,Male_OC$ClassID)\n",
    "Male_CO_granges <- GRanges(seqnames=Male_CO$seqname,ranges=IRanges(start = Male_CO$Start,end = Male_CO$End, names = Male_CO$GeneID),strand=NULL,Male_CO$DupID,Male_CO$FamilyID,Male_CO$ClassID)\n",
    "#\n",
    "Female_OC_granges <- GRanges(seqnames=Female_OC$seqname,ranges=IRanges(start = Female_OC$Start,end = Female_OC$End, names = Female_OC$GeneID),strand=NULL,Female_OC$DupID,Female_OC$FamilyID,Female_OC$ClassID)\n",
    "Female_CO_granges <- GRanges(seqnames=Female_CO$seqname,ranges=IRanges(start = Female_CO$Start,end = Female_CO$End, names = Female_CO$GeneID),strand=NULL,Female_CO$DupID,Female_CO$FamilyID,Female_CO$ClassID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "153a453a",
   "metadata": {},
   "outputs": [],
   "source": [
    "p <- c(\"d6_peak_granges\")\n",
    "a <- c(\"Male_OC_granges\")\n",
    "\n",
    "peak <- get(p)\n",
    "ATAC <- get(a)\n",
    "\n",
    "##This code inspired by a GitHub script (https://gist.githubusercontent.com/ATpoint/0c05bf59b8c80803a525494e63e05e8e/raw/dbac4ccc6b535e5d7e0174c119fa51a9e8568444/GetPercentOverlap.R)\n",
    "##hosted by https://gist.github.com/ATpoint/)\n",
    "overlap <- findOverlaps(peak, ATAC)\n",
    "#Takes matching ranges from overlap object and stores them as IR Range Objects\n",
    "r1 <- ranges(peak[overlap@from] )\n",
    "r2 <- ranges(ATAC[overlap@to])\n",
    "#Merges r1 and r2, and for the overlap takes the largest start and the smallest end points, which are the most\n",
    "#conservative peak sizes.\n",
    "gr<-\n",
    "  GRanges(seqnames = as.character(seqnames(peak[overlap@from])),\n",
    "          ranges = IRanges(start = matrixStats::rowMaxs(as.matrix(data.frame(r1=start(r1), r2=start(r2)))),\n",
    "                           end = matrixStats::rowMins(as.matrix(data.frame(r1=end(r1), r2=end(r2))))))\n",
    "#This is \"width of overlap\" over \"width of peak\"\n",
    "mcols(gr)$percentOverlap <- 100 * width(gr) / width(r1)\n",
    "#Takes metadata and ranges from original object filtered for overlaps and replaces them in gr object\n",
    "ranges(gr) <- ranges(r1)\n",
    "#Adds metadata from r2\n",
    "mcols(gr)$subject <- r2\n",
    "gr <- data.frame(gr)\n",
    "out <- filter(gr, percentOverlap >= 20)\n",
    "Percent_Overlap <- data.frame(Contrast = (paste(p, a, sep = \"*\")), Percent=((nrow(out)/nrow(data.frame(ATAC)))*100))#, stringsAsFactors=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "f659a268",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Contrast</th><th scope=col>Percent</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>d6_peak_granges*Male_OC_granges  </td><td>13.553613</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_OC_granges</td><td> 8.128489</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Contrast & Percent\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t d6\\_peak\\_granges*Male\\_OC\\_granges   & 13.553613\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_OC\\_granges &  8.128489\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| Contrast &lt;chr&gt; | Percent &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| d6_peak_granges*Male_OC_granges   | 13.553613 |\n",
       "| d6_peak_granges*Female_OC_granges |  8.128489 |\n",
       "\n"
      ],
      "text/plain": [
       "  Contrast                          Percent  \n",
       "1 d6_peak_granges*Male_OC_granges   13.553613\n",
       "2 d6_peak_granges*Female_OC_granges  8.128489"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p <- c(\"d6_peak_granges\")\n",
    "a <- c(\"Female_OC_granges\")\n",
    "\n",
    "peak <- get(p)\n",
    "ATAC <- get(a)\n",
    "\n",
    "##This code inspired by a GitHub script (https://gist.githubusercontent.com/ATpoint/0c05bf59b8c80803a525494e63e05e8e/raw/dbac4ccc6b535e5d7e0174c119fa51a9e8568444/GetPercentOverlap.R)\n",
    "##hosted by https://gist.github.com/ATpoint/)\n",
    "overlap <- findOverlaps(peak, ATAC)\n",
    "#Takes matching ranges from overlap object and stores them as IR Range Objects\n",
    "r1 <- ranges(peak[overlap@from] )\n",
    "r2 <- ranges(ATAC[overlap@to])\n",
    "#Merges r1 and r2, and for the overlap takes the largest start and the smallest end points, which are the most\n",
    "#conservative peak sizes.\n",
    "gr<-\n",
    "  GRanges(seqnames = as.character(seqnames(peak[overlap@from])),\n",
    "          ranges = IRanges(start = matrixStats::rowMaxs(as.matrix(data.frame(r1=start(r1), r2=start(r2)))),\n",
    "                           end = matrixStats::rowMins(as.matrix(data.frame(r1=end(r1), r2=end(r2))))))\n",
    "#This is \"width of overlap\" over \"width of peak\"\n",
    "mcols(gr)$percentOverlap <- 100 * width(gr) / width(r1)\n",
    "#Takes metadata and ranges from original object filtered for overlaps and replaces them in gr object\n",
    "ranges(gr) <- ranges(r1)\n",
    "#Adds metadata from r2\n",
    "mcols(gr)$subject <- r2\n",
    "gr <- data.frame(gr)\n",
    "out <- filter(gr, percentOverlap >= 20)\n",
    "Perc <- data.frame(Contrast = (paste(p, a, sep = \"*\")), Percent=((nrow(out)/nrow(data.frame(ATAC)))*100))#, stringsAsFactors=FALSE)\n",
    "Percent_Overlap <- rbind(Percent_Overlap, Perc)\n",
    "Percent_Overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "069bba1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Contrast</th><th scope=col>Percent</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>d6_peak_granges*Male_OC_granges  </td><td>13.553613</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_OC_granges</td><td> 8.128489</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Male_CO_granges  </td><td> 4.808048</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Contrast & Percent\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t d6\\_peak\\_granges*Male\\_OC\\_granges   & 13.553613\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_OC\\_granges &  8.128489\\\\\n",
       "\t d6\\_peak\\_granges*Male\\_CO\\_granges   &  4.808048\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 2\n",
       "\n",
       "| Contrast &lt;chr&gt; | Percent &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| d6_peak_granges*Male_OC_granges   | 13.553613 |\n",
       "| d6_peak_granges*Female_OC_granges |  8.128489 |\n",
       "| d6_peak_granges*Male_CO_granges   |  4.808048 |\n",
       "\n"
      ],
      "text/plain": [
       "  Contrast                          Percent  \n",
       "1 d6_peak_granges*Male_OC_granges   13.553613\n",
       "2 d6_peak_granges*Female_OC_granges  8.128489\n",
       "3 d6_peak_granges*Male_CO_granges    4.808048"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p <- c(\"d6_peak_granges\")\n",
    "a <- c(\"Male_CO_granges\")\n",
    "\n",
    "peak <- get(p)\n",
    "ATAC <- get(a)\n",
    "\n",
    "##This code inspired by a GitHub script (https://gist.githubusercontent.com/ATpoint/0c05bf59b8c80803a525494e63e05e8e/raw/dbac4ccc6b535e5d7e0174c119fa51a9e8568444/GetPercentOverlap.R)\n",
    "##hosted by https://gist.github.com/ATpoint/)\n",
    "overlap <- findOverlaps(peak, ATAC)\n",
    "#Takes matching ranges from overlap object and stores them as IR Range Objects\n",
    "r1 <- ranges(peak[overlap@from] )\n",
    "r2 <- ranges(ATAC[overlap@to])\n",
    "#Merges r1 and r2, and for the overlap takes the largest start and the smallest end points, which are the most\n",
    "#conservative peak sizes.\n",
    "gr<-\n",
    "  GRanges(seqnames = as.character(seqnames(peak[overlap@from])),\n",
    "          ranges = IRanges(start = matrixStats::rowMaxs(as.matrix(data.frame(r1=start(r1), r2=start(r2)))),\n",
    "                           end = matrixStats::rowMins(as.matrix(data.frame(r1=end(r1), r2=end(r2))))))\n",
    "#This is \"width of overlap\" over \"width of peak\"\n",
    "mcols(gr)$percentOverlap <- 100 * width(gr) / width(r1)\n",
    "#Takes metadata and ranges from original object filtered for overlaps and replaces them in gr object\n",
    "ranges(gr) <- ranges(r1)\n",
    "#Adds metadata from r2\n",
    "mcols(gr)$subject <- r2\n",
    "gr <- data.frame(gr)\n",
    "out <- filter(gr, percentOverlap >= 20)\n",
    "Perc <- data.frame(Contrast = (paste(p, a, sep = \"*\")), Percent=((nrow(out)/nrow(data.frame(ATAC)))*100))#, stringsAsFactors=FALSE)\n",
    "Percent_Overlap <- rbind(Percent_Overlap, Perc)\n",
    "Percent_Overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "ceb4264f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Contrast</th><th scope=col>Percent</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>d6_peak_granges*Male_OC_granges  </td><td>13.553613</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_OC_granges</td><td> 8.128489</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Male_CO_granges  </td><td> 4.808048</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_CO_granges</td><td> 5.652006</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Contrast & Percent\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t d6\\_peak\\_granges*Male\\_OC\\_granges   & 13.553613\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_OC\\_granges &  8.128489\\\\\n",
       "\t d6\\_peak\\_granges*Male\\_CO\\_granges   &  4.808048\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_CO\\_granges &  5.652006\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 2\n",
       "\n",
       "| Contrast &lt;chr&gt; | Percent &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| d6_peak_granges*Male_OC_granges   | 13.553613 |\n",
       "| d6_peak_granges*Female_OC_granges |  8.128489 |\n",
       "| d6_peak_granges*Male_CO_granges   |  4.808048 |\n",
       "| d6_peak_granges*Female_CO_granges |  5.652006 |\n",
       "\n"
      ],
      "text/plain": [
       "  Contrast                          Percent  \n",
       "1 d6_peak_granges*Male_OC_granges   13.553613\n",
       "2 d6_peak_granges*Female_OC_granges  8.128489\n",
       "3 d6_peak_granges*Male_CO_granges    4.808048\n",
       "4 d6_peak_granges*Female_CO_granges  5.652006"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p <- c(\"d6_peak_granges\")\n",
    "a <- c(\"Female_CO_granges\")\n",
    "\n",
    "peak <- get(p)\n",
    "ATAC <- get(a)\n",
    "\n",
    "##This code inspired by a GitHub script (https://gist.githubusercontent.com/ATpoint/0c05bf59b8c80803a525494e63e05e8e/raw/dbac4ccc6b535e5d7e0174c119fa51a9e8568444/GetPercentOverlap.R)\n",
    "##hosted by https://gist.github.com/ATpoint/)\n",
    "overlap <- findOverlaps(peak, ATAC)\n",
    "#Takes matching ranges from overlap object and stores them as IR Range Objects\n",
    "r1 <- ranges(peak[overlap@from] )\n",
    "r2 <- ranges(ATAC[overlap@to])\n",
    "#Merges r1 and r2, and for the overlap takes the largest start and the smallest end points, which are the most\n",
    "#conservative peak sizes.\n",
    "gr<-\n",
    "  GRanges(seqnames = as.character(seqnames(peak[overlap@from])),\n",
    "          ranges = IRanges(start = matrixStats::rowMaxs(as.matrix(data.frame(r1=start(r1), r2=start(r2)))),\n",
    "                           end = matrixStats::rowMins(as.matrix(data.frame(r1=end(r1), r2=end(r2))))))\n",
    "#This is \"width of overlap\" over \"width of peak\"\n",
    "mcols(gr)$percentOverlap <- 100 * width(gr) / width(r1)\n",
    "#Takes metadata and ranges from original object filtered for overlaps and replaces them in gr object\n",
    "ranges(gr) <- ranges(r1)\n",
    "#Adds metadata from r2\n",
    "mcols(gr)$subject <- r2\n",
    "gr <- data.frame(gr)\n",
    "out <- filter(gr, percentOverlap >= 20)\n",
    "Perc <- data.frame(Contrast = (paste(p, a, sep = \"*\")), Percent=((nrow(out)/nrow(data.frame(ATAC)))*100))#, stringsAsFactors=FALSE)\n",
    "Percent_Overlap <- rbind(Percent_Overlap, Perc)\n",
    "Percent_Overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "b46fd252",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 5 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Contrast</th><th scope=col>Percent</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>d6_peak_granges*Male_OC_granges  </td><td>13.553613</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_OC_granges</td><td> 8.128489</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Male_CO_granges  </td><td> 4.808048</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_CO_granges</td><td> 5.652006</td></tr>\n",
       "\t<tr><td>E135_M_granges*Male_CO_granges   </td><td> 4.326626</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Contrast & Percent\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t d6\\_peak\\_granges*Male\\_OC\\_granges   & 13.553613\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_OC\\_granges &  8.128489\\\\\n",
       "\t d6\\_peak\\_granges*Male\\_CO\\_granges   &  4.808048\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_CO\\_granges &  5.652006\\\\\n",
       "\t E135\\_M\\_granges*Male\\_CO\\_granges    &  4.326626\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 2\n",
       "\n",
       "| Contrast &lt;chr&gt; | Percent &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| d6_peak_granges*Male_OC_granges   | 13.553613 |\n",
       "| d6_peak_granges*Female_OC_granges |  8.128489 |\n",
       "| d6_peak_granges*Male_CO_granges   |  4.808048 |\n",
       "| d6_peak_granges*Female_CO_granges |  5.652006 |\n",
       "| E135_M_granges*Male_CO_granges    |  4.326626 |\n",
       "\n"
      ],
      "text/plain": [
       "  Contrast                          Percent  \n",
       "1 d6_peak_granges*Male_OC_granges   13.553613\n",
       "2 d6_peak_granges*Female_OC_granges  8.128489\n",
       "3 d6_peak_granges*Male_CO_granges    4.808048\n",
       "4 d6_peak_granges*Female_CO_granges  5.652006\n",
       "5 E135_M_granges*Male_CO_granges     4.326626"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p <- c(\"E135_M_granges\")\n",
    "a <- c(\"Male_CO_granges\")\n",
    "\n",
    "peak <- get(p)\n",
    "ATAC <- get(a)\n",
    "\n",
    "##This code inspired by a GitHub script (https://gist.githubusercontent.com/ATpoint/0c05bf59b8c80803a525494e63e05e8e/raw/dbac4ccc6b535e5d7e0174c119fa51a9e8568444/GetPercentOverlap.R)\n",
    "##hosted by https://gist.github.com/ATpoint/)\n",
    "overlap <- findOverlaps(peak, ATAC)\n",
    "#Takes matching ranges from overlap object and stores them as IR Range Objects\n",
    "r1 <- ranges(peak[overlap@from] )\n",
    "r2 <- ranges(ATAC[overlap@to])\n",
    "#Merges r1 and r2, and for the overlap takes the largest start and the smallest end points, which are the most\n",
    "#conservative peak sizes.\n",
    "gr<-\n",
    "  GRanges(seqnames = as.character(seqnames(peak[overlap@from])),\n",
    "          ranges = IRanges(start = matrixStats::rowMaxs(as.matrix(data.frame(r1=start(r1), r2=start(r2)))),\n",
    "                           end = matrixStats::rowMins(as.matrix(data.frame(r1=end(r1), r2=end(r2))))))\n",
    "#This is \"width of overlap\" over \"width of peak\"\n",
    "mcols(gr)$percentOverlap <- 100 * width(gr) / width(r1)\n",
    "#Takes metadata and ranges from original object filtered for overlaps and replaces them in gr object\n",
    "ranges(gr) <- ranges(r1)\n",
    "#Adds metadata from r2\n",
    "mcols(gr)$subject <- r2\n",
    "gr <- data.frame(gr)\n",
    "out <- filter(gr, percentOverlap >= 20)\n",
    "Perc <- data.frame(Contrast = (paste(p, a, sep = \"*\")), Percent=((nrow(out)/nrow(data.frame(ATAC)))*100))#, stringsAsFactors=FALSE)\n",
    "Percent_Overlap <- rbind(Percent_Overlap, Perc)\n",
    "Percent_Overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "b3da95fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Contrast</th><th scope=col>Percent</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>d6_peak_granges*Male_OC_granges  </td><td>13.553613</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_OC_granges</td><td> 8.128489</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Male_CO_granges  </td><td> 4.808048</td></tr>\n",
       "\t<tr><td>d6_peak_granges*Female_CO_granges</td><td> 5.652006</td></tr>\n",
       "\t<tr><td>E135_M_granges*Male_CO_granges   </td><td> 4.326626</td></tr>\n",
       "\t<tr><td>E135_F_granges*Female_CO_granges </td><td> 1.909722</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Contrast & Percent\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t d6\\_peak\\_granges*Male\\_OC\\_granges   & 13.553613\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_OC\\_granges &  8.128489\\\\\n",
       "\t d6\\_peak\\_granges*Male\\_CO\\_granges   &  4.808048\\\\\n",
       "\t d6\\_peak\\_granges*Female\\_CO\\_granges &  5.652006\\\\\n",
       "\t E135\\_M\\_granges*Male\\_CO\\_granges    &  4.326626\\\\\n",
       "\t E135\\_F\\_granges*Female\\_CO\\_granges  &  1.909722\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| Contrast &lt;chr&gt; | Percent &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| d6_peak_granges*Male_OC_granges   | 13.553613 |\n",
       "| d6_peak_granges*Female_OC_granges |  8.128489 |\n",
       "| d6_peak_granges*Male_CO_granges   |  4.808048 |\n",
       "| d6_peak_granges*Female_CO_granges |  5.652006 |\n",
       "| E135_M_granges*Male_CO_granges    |  4.326626 |\n",
       "| E135_F_granges*Female_CO_granges  |  1.909722 |\n",
       "\n"
      ],
      "text/plain": [
       "  Contrast                          Percent  \n",
       "1 d6_peak_granges*Male_OC_granges   13.553613\n",
       "2 d6_peak_granges*Female_OC_granges  8.128489\n",
       "3 d6_peak_granges*Male_CO_granges    4.808048\n",
       "4 d6_peak_granges*Female_CO_granges  5.652006\n",
       "5 E135_M_granges*Male_CO_granges     4.326626\n",
       "6 E135_F_granges*Female_CO_granges   1.909722"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p <- c(\"E135_F_granges\")\n",
    "a <- c(\"Female_CO_granges\")\n",
    "\n",
    "peak <- get(p)\n",
    "ATAC <- get(a)\n",
    "\n",
    "##This code inspired by a GitHub script (https://gist.githubusercontent.com/ATpoint/0c05bf59b8c80803a525494e63e05e8e/raw/dbac4ccc6b535e5d7e0174c119fa51a9e8568444/GetPercentOverlap.R)\n",
    "##hosted by https://gist.github.com/ATpoint/)\n",
    "overlap <- findOverlaps(peak, ATAC)\n",
    "#Takes matching ranges from overlap object and stores them as IR Range Objects\n",
    "r1 <- ranges(peak[overlap@from] )\n",
    "r2 <- ranges(ATAC[overlap@to])\n",
    "#Merges r1 and r2, and for the overlap takes the largest start and the smallest end points, which are the most\n",
    "#conservative peak sizes.\n",
    "gr<-\n",
    "  GRanges(seqnames = as.character(seqnames(peak[overlap@from])),\n",
    "          ranges = IRanges(start = matrixStats::rowMaxs(as.matrix(data.frame(r1=start(r1), r2=start(r2)))),\n",
    "                           end = matrixStats::rowMins(as.matrix(data.frame(r1=end(r1), r2=end(r2))))))\n",
    "#This is \"width of overlap\" over \"width of peak\"\n",
    "mcols(gr)$percentOverlap <- 100 * width(gr) / width(r1)\n",
    "#Takes metadata and ranges from original object filtered for overlaps and replaces them in gr object\n",
    "ranges(gr) <- ranges(r1)\n",
    "#Adds metadata from r2\n",
    "mcols(gr)$subject <- r2\n",
    "gr <- data.frame(gr)\n",
    "out <- filter(gr, percentOverlap >= 20)\n",
    "Perc <- data.frame(Contrast = (paste(p, a, sep = \"*\")), Percent=((nrow(out)/nrow(data.frame(ATAC)))*100))#, stringsAsFactors=FALSE)\n",
    "Percent_Overlap <- rbind(Percent_Overlap, Perc)\n",
    "Percent_Overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "c48ad1cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(Percent_Overlap, file = 'Percent_Overlap_ATAC_H3K27ac.tsv', sep = \"\\t\", quote = F, col.names = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e93c7129",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.2.1",
   "language": "R",
   "name": "ir421"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
